tests/test_builder.py

```diff
@@ -777,5 +777,80 @@ def get_mlir_string(schedule):
     assert 'loc("' in mlir_string
 
 
+def test_dsl_broadcast_add():
+    def kernel1(A: int32[10]) -> int32[10]:
+        return allo.add(A, 1)
+
+    def kernel2(A: int32[10]) -> int32[10]:
+        return allo.add(1, A)
```

Comment:
Can you use more diverse and complicated test cases? For example, `(allo.add(A, 1) - 2)*3`, which can test multiple features in one case.